### ISSUE TITLE
False positive with rayon's `into_par_iter`, `fold` and `reduce`

### ISSUE THREAD

**Issue by jorgecarleitao** [rust-analyzer/rust-analyzer#10952]
Since I am at it, thanks a lot for this amazing piece of software!

Minimal repro I could come up with:

```rust
use std::collections::HashMap;

use rayon::prelude::*;

type C = HashMap<u64, (Vec<u64>, Vec<u64>)>;

fn main() {
    let a = vec![1u64];
    a.into_par_iter()
        .enumerate()
        .fold(C::default, |_, _| todo!())
        .reduce(C::default, |acc, other| todo!());
}
```

it shows as an error in VS code with the configuration below, but it compiles fine. Changing to `type C = HashMap<u64, (u64, u64)>;` does not show the error.

**rust-analyzer version**: `rust-analyzer version: 7d6fcbc0b 2021-12-06 stable`

**rustc version**: `rustc 1.56.1 (59eed8a2a 2021-11-01)`

**Comment by martinxyz** [comment]
Confirmed. Most likely a duplicate of #10822. According to the help, it picks `reduce` from the `Iterator` trait instead of `Paralllelterator`.

**Comment by lnicola** [comment]
Probably fixed in #11074:

![image](https://user-images.githubusercontent.com/308347/146896494-e8033e77-1d7c-4bec-adb9-f3b90f72181b.png)
